src/main_wrapper.c

```diff
@@ -13,288 +13,300 @@
+      {"luagit2_branch_next", lua_git_branch_next},
+      {"luagit2_branch_set_upstream", lua_git_branch_set_upstream},
+      {"luagit2_branch_upstream", lua_git_branch_upstream},
       //-----------------------------------------------------------------------------------------------//
 
       //------------------------------------------- Blame Methods -------------------------------------//
-      {"luagit2_blame_buffer",lua_git_blame_buffer},
-      {"luagit2_blame_file",lua_git_blame_file},
-      {"luagit2_blame_get_hunk_byindex",lua_git_blame_get_hunk_byindex},
-      {"luagit2_blame_get_hunk_byline",lua_git_blame_get_hunk_byline},
-      {"luagit2_blame_get_hunk_count",lua_git_blame_get_hunk_count},
+      {"luagit2_blame_buffer", lua_git_blame_buffer},
+      {"luagit2_blame_file", lua_git_blame_file},
+      {"luagit2_blame_get_hunk_byindex", lua_git_blame_get_hunk_byindex},
+      {"luagit2_blame_get_hunk_byline", lua_git_blame_get_hunk_byline},
+      {"luagit2_blame_get_hunk_count", lua_git_blame_get_hunk_count},
+      {"luagit2_blame_free", lua_git_blame_free},
       //-----------------------------------------------------------------------------------------------//
 
       //------------------------------------------ Blob Methods ---------------------------------------//
-      {"luagit2_blob_create_fromdisk",lua_git_blob_create_fromdisk},
-      {"luagit2_blob_create_fromworkdir",lua_git_blob_create_fromworkdir},
-      {"luagit2_blob_filtered_content",lua_git_blob_filtered_content},
-      {"luagit2_blob_id",lua_git_blob_id},
-      {"luagit2_blob_is_binary",lua_git_blob_is_binary},
-      {"luagit2_blob_lookup",lua_git_blob_lookup},
-      {"luagit2_blob_lookup_prefix",lua_git_blob_lookup_prefix},
-      {"luagit2_blob_owner",lua_git_blob_owner},
-      {"luagit2_blob_rawsize",lua_git_blob_rawsize},
+      {"luagit2_blob_create_fromdisk", lua_git_blob_create_fromdisk},
+      {"luagit2_blob_create_fromworkdir", lua_git_blob_create_fromworkdir},
+      {"luagit2_blob_filtered_content", lua_git_blob_filtered_content},
+      {"luagit2_blob_id", lua_git_blob_id},
+      {"luagit2_blob_is_binary", lua_git_blob_is_binary},
+      {"luagit2_blob_lookup", lua_git_blob_lookup},
+      {"luagit2_blob_lookup_prefix", lua_git_blob_lookup_prefix},
+      {"luagit2_blob_owner", lua_git_blob_owner},
+      {"luagit2_blob_rawsize", lua_git_blob_rawsize},
+      {"luagit2_blob_free", lua_git_blob_free},
       //-----------------------------------------------------------------------------------------------//
 
       //------------------------------------------- Helper Methods ------------------------------------//
-      {"luagit2_print_version_data",lua_git_libgit2_print_version_data},
-      {"luagit2_get_signature_details",lua_get_signature_details},
-      {"luagit2_print_complete_signature_details",lua_print_complete_signature_details},
+      {"luagit2_print_version_data", lua_git_libgit2_print_version_data},
```

Comment:
If I understood correctly, these methods do not belong originally to libgit2, correct? Are they functions you made especially for luagit2?

Reply:
Yes, I was unable to access data from actual c struct objects, I therefore Created them to get those data.

Reply:
ok!